### ISSUE TITLE
IE11 support, arrow functions & backticks

### ISSUE THREAD

**Issue by kbjerke** [CanopyTax/single-spa-angular#56]
Hi, this project is incompatible with IE11, and if I build it using target es5, the builders do not work. Any dependency between the two or could it be separated into different projects? I understand that this probably isn't part of your requirement.

**Comment by joeldenning** [comment]
Hi @kbjerke, IE11 support is something that we want to have as part of single-spa-angular.

Could you provide more detail about which error you are seeing? Maybe a stacktrace or screenshot when you don't target es5 and another one when you do?

**Comment by jlwerenskjold** [comment]
I've been working on trying to get it working in IE 11 today, no luck so far.  I've been using this project https://github.com/joeldenning/coexisting-angular-microfrontends to test solutions.  I've added the following polyfill imports as was suggested online, but it almost seems like it is not picking them up.  I'm not exactly sure how the polyfill works so I'm looking into that now.

import 'core-js/es6/symbol'; 
import 'core-js/es6/object'; 
import 'core-js/es6/function'; 
import 'core-js/es6/parse-int'; 
import 'core-js/es6/parse-float';
import 'core-js/es6/number';
import 'core-js/es6/math';
import 'core-js/es6/string';
import 'core-js/es6/date';
import 'core-js/es6/array';
import 'core-js/es7/array';
import 'core-js/es6/regexp';
import 'core-js/es6/map';
import 'core-js/es6/weak-map';
import 'core-js/es6/set';
import 'core-js/es6/reflect';

I've also messed around with importing some shim-es5 scripts, setting the compatibility mode to edge, and other kinda silly solutions with no success.  I've attached some screenshots of the errors, it just seems that IE does not recognize some of the javascript syntax.

I'm currently targeting es5.

![2019-04-11 14_03_23-](https://user-images.githubusercontent.com/6802821/55980680-fea09d00-5c62-11e9-8143-fa30dc1fd08d.png)
![2019-04-11 14_04_02-](https://user-images.githubusercontent.com/6802821/55980686-02ccba80-5c63-11e9-84f8-89428d244a2d.png)
![2019-04-11 14_04_27-](https://user-images.githubusercontent.com/6802821/55980692-05c7ab00-5c63-11e9-8b0f-abe073b79775.png)
![2019-04-11 14_07_25-](https://user-images.githubusercontent.com/6802821/55980775-33acef80-5c63-11e9-84e0-668192fe2082.png)

**Comment by joeldenning** [comment]
Ah I see. Yes. Polyfills are not being loaded when you are using single-spa-angular, although that is intentional. See https://github.com/CanopyTax/single-spa-angular/issues/35#issuecomment-480467944

The reason is that a big difference between two angular apps using single-spa versus two that are not is that **when using single-spa they share an html file, instead of having their own html file**. How this relates to polyfills is that when you have multiple angular applications, you don't want to have multiple instances of the polyfills loaded into that one shared html file file. Instead, polyfills should be loaded once upfront in the single-spa config html file.

So what you'll want to do is load up core-js inside of the single-spa config html file and then each of the angular apps won't have to worry about polyfills anymore. To do that, try adding `<script src="https://cdnjs.cloudflare.com/ajax/libs/core-js/2.6.5/core.min.js"></script>` to your single-spa config html file. See https://cdnjs.com/libraries/core-js for more info

**Comment by kbjerke** [comment]
@joeldenning , Thank you for the quick response, and sorry for my lacking information. 

Yes, polyfills was a part of the issue, but sadly not related to my report. 

Approach:
With https://github.com/joeldenning/coexisting-angular-microfrontends as my base, I set the target of  App1, App2 and Navbar to es5, in tsconfig.json under compilerOptions. Now what I was seeing in the bundle was that the singleSpaAngular function was bundled with es6 features like arrow functions and backticks. That's what causing the Syntax errors, as there's no polyfill for those. I made sure to import: 
`<script src="https://unpkg.com/core-js-bundle@3.0.1/minified.js"></script>`
before zone.js in the index.html.

examples from navbar/dist/navbar/main.js:

```
if (!opts.NgZone) {
    throw Error(`single-spa-angular must be passed the NgZone opt`);
}
...
return Promise.resolve().then(() => {
```
etc.

1)
So the first approach was to clone the master branch of this project, rebuild it using es5 as target, and installing it from filesystem.
```
PS C:\projects\coexisting-angular-microfrontends-master\navbar> ng build
Class constructor BrowserBuilder cannot be invoked without 'new'
TypeError: Class constructor BrowserBuilder cannot be invoked without 'new'
    at new SingleSpaBrowserBuilder (C:\projects\single-spa-angular-master\lib\builders\browser\index.js:21:42)
    at Architect.getBuilder (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\@angular-devkit\architect\src\architect-legacy.js:215:25)
    at MapSubscriber.getBuilderDescription.pipe.operators_1.map [as project] (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\@angular-devkit\architect\src\architect-legacy.js:142:332)
    at MapSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\map.js:49:35)
    at MapSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at TapSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\tap.js:65:26)
    at TapSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at MergeMapSubscriber.notifyNext (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:92:26)
    at InnerSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\InnerSubscriber.js:28:21)
    at InnerSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at MapSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\map.js:55:26)
    at MapSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at MergeMapSubscriber.notifyNext (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:92:26)
    at InnerSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\InnerSubscriber.js:28:21)
    at InnerSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\util\subscribeTo.js:17:28
    at Object.subscribeToResult (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\util\subscribeToResult.js:10:45)
    at MergeMapSubscriber._innerSub (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:82:29)
    at MergeMapSubscriber._tryNext (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:76:14)
    at MergeMapSubscriber._next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:59:18)
    at MergeMapSubscriber.Subscriber.next (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\Subscriber.js:67:18)
    at MergeMapSubscriber.notifyNext (C:\projects\coexisting-angular-microfrontends-master\navbar\node_modules\rxjs\internal\operators\mergeMap.js:92:26)
```
So now the build process isn't working for the projects.

2)
My second approach was basically copying the single-spa-angular.ts file into my projects, and now it works fine. (I was lazy, I didn't want to create a project for it) And I'm using the builders from the package.

In Edge, Chrome and Firefox the projects work fine, but in IE11:
![image](https://user-images.githubusercontent.com/35374983/56016403-6bb44100-5cfc-11e9-9e3c-755d70354d11.png)

**Comment by jlwerenskjold** [comment]
@kbjerke I got it working in IE 11 by removing the backticks and converting to anonymous functions in the single-spa-angular.js file.  If you modified single-spa-angular, I highly suggest using https://github.com/ds300/patch-package as it is a wonderul tool for this.  Basically you:

1. Modify the single-spa-angular,js file in your node_modules/single-spa-angular/lib folder by replacing the backticks to quotes and lambdas to anonymous functions.
2. Run 'npm install patch-package', I saved it as a dev-dependency.
3. Run 'npx patch-package single-spa-angular' which will create a patch file for you.
4. Add "postinstall": "patch-package" to your package.json scripts, which will apply the patch to single-spa-angular when you run npm install.

This, along with including core.js into the index config file is what worked for me.

**Comment by joeldenning** [comment]
Oh I didn't realize that the backticks and arrow functions weren't being compiled down. We recently switched from babel to typescript in #23, and must have introduced this problem at that time.

I will push a fix!

**Comment by joeldenning** [comment]
This is fixed in [3.0.0-beta.17](https://github.com/CanopyTax/single-spa-angular/releases/tag/v3.0.0-beta.17)

Note how [the published package](https://unpkg.com/single-spa-angular@3.0.0-beta.17/lib/browser-lib/single-spa-angular.js) now does not have any backticks or arrow functions anymore.

**Comment by hiskuDN** [comment]
I'm getting the same error on IE 11. The images show the single-spa-config.js file and the error produced on IE 11
![image](https://user-images.githubusercontent.com/18376750/100326125-ea164180-300c-11eb-84f8-6ea6085fa5df.png)
![image](https://user-images.githubusercontent.com/18376750/100326179-fa2e2100-300c-11eb-9014-3486bfbef7ce.png)
